docs/nebula-design-concepts.md

```diff
@@ -0,0 +1,214 @@
+
+In the preceding example, there are two kinds of edges, one is **like** pointing
+from player to player, the property is likeness (double); the other is **serve**,
+pointing from team to serve, the properties are start_year(int) and end_year(int).
+
+It should be noted that there may be **multiple edges of the same or different types
+at the same time** between the start (src)and end vertex (dst).
+
+
+
+## Graph partition
+
+![image](https://user-images.githubusercontent.com/42762957/61119934-76829b80-a4ce-11e9-9d49-2abb11b5f7b2.png)
+
+
+In applications such as social networks, or road networks, the number of vertices and their connections are extremely large. Therefore, partitioning large graphs is inevitable for complexity reduction and parallelization. Nebula Graph adopts **Edge Separators** to cut a graph into smaller components. And the partition number cannot be changed once being configured.
+
+## Data model
+
+
+![image](https://user-images.githubusercontent.com/42762957/61120073-b0ec3880-a4ce-11e9-975f-c19482d4b109.png)
+
+
+
+Data is stored as key-value pairs in Nebula Graph. For vertices, as illustrated in the diagram above, the key consists of four parts: partID (4 bit), vertexID (8 bit), tagID (4 bit) and Version (8 bit). And they are spread across various partitions based on the hash of their vertexIds (vid).
+
+For edges, the key consists of two separate key-values, i.e. out-key and
+in-key, respectively. The out-key and the edge's corresponding starting point are stored in one partition, while the in-key and end point are stored in another one.
+
+
+
+![image](https://user-images.githubusercontent.com/42762957/61120260-1c360a80-a4cf-11e9-8a43-8c4ca2d73572.png)
+
+
+
+## Architecture
+
+Nebula Graph includes four modules: storage service, metadata service, query engine and client.
+
+
+![image](https://user-images.githubusercontent.com/42762957/61120288-31ab3480-a4cf-11e9-9905-a1d4b1e6c523.png)
+
+
+
+
+### Storage serve
+
+The corresponding process of storage service is nebula-storaged, and provides a key-value store. Separated storage and computing makes storage service flexible, and multiple storage engines like RocksDB and HBase are supported, with RocksDB set as default engine. To build a resilient distributed system, [Raft](https://raft.github.io/) is implemented as the consensus algorithm.
+
+Currently it supports
+multiple storage engines like Rocksdb and HBase, etc.
+
+Raft achieves data consensus via an elected leader. Based on that, Nebula Storage makes the following optimizations:
+
+- Parallel Raft
+
+      Partitions of same ID from multiple machines form a raft group. And the concurrent operations are implemented with multiple sets of Raft groups.
```

Comment:
In Raft protocol, multi-machine synchronization for one raft group depends on log ID orders, which
   lead to low throughput.  So we support that one host could hold many raft groups.  And the operations on different group could be invoked simultaneously.